net/sync/service/src/main/java/de/danoeh/antennapod/net/sync/service/SyncService.java

```diff
@@ -201,6 +201,12 @@ private void syncSubscriptions(ISyncService syncServiceImpl) throws SyncServiceE
                         .uploadSubscriptionChanges(queuedAddedFeeds, queuedRemovedFeeds);
                 synchronizationQueueStorage.clearFeedQueues();
                 newTimeStamp = uploadResponse.timestamp;
+            } catch (SyncServiceException exception) {
+                // In some cases, syncing with gpodder will fail repeatedly because there is legacy data in the
+                // synchronization storage containing the same feed url as "added" and "removed".
+                // If synchronization fails, clean up any conflicts to make sure that when it fails
+                // for this reason the next sync can succeed.
+                synchronizationQueueStorage.removeLegacyConflictingFeedEntries(localSubscriptions);
```

Comment:
Nice idea doing this only when sync fails. Some comments:

- Your current implementation swallows the exception. This means that the code will continue to run and try to sync episodes for subscriptions that might not exist yet. So the exception still needs to be thrown here.
- the comment here is quite detailed. I think it is not needed here, given that the method name and method documentation already explain this. This makes the overall class a bit easier to read when looking at the "non-error" case.